### ISSUE TITLE
How should we handle the loading of commands?

### ISSUE THREAD

**Issue by adigitalmonk** [adigitalmonk/durendal-bot#51]
Right now we are handling it with a try/catch which is kind of dirty.

One idea I had was we could create some kind of cache somewhere of mappings for 'command' => require('some_command.js').  This would give us more freedom over naming commands and we would not have to worry about exceptions every time someone types a command that doesn't exist.  Not sure where we'd hold that static object, maybe in the bootstrap object?

**Comment by dmasterson** [comment]
I can see utility in knowing about all of our commands up front. However, what do we do about adding new commands? Upgrading commands? Do we force a reboot of the bot? Should we move to a model like commands-available/commands-enabled?

**Comment by adigitalmonk** [comment]
So I'm thinking it's not unreasonable to reboot the bot when we do a code push.  The benefit of being able to load new commands live isn't necessarily a deal breaker as I think about it.

Ideally we'd have some kind of commands mappings (probably in the listener for commands? or maybe something global?)

e.g.,
```
// either something global, something configured?
const cmd_map = {
    'dice' => 'src/commands/dice.js'
};
// ...
if (cmd_map[dice]) {
     let cmd = require(cmd_map[dice]);
}
```
This could be something included in that `global.js` conversation we had.

**Comment by adigitalmonk** [comment]
Also, maybe we could use this to configure functionality about specific commands?  Restricted could be stored in the map, along with how many times until it gets throttled?

```
const cmd_map = {
    'dice' => {
         'src' => 'src/commands/dice.js',
         'restricted' => false,
         'limit' => 20
     }
};
```
?

**Comment by adigitalmonk** [comment]
I created a branch where I implemented something like this using some global methods.  I'm trying to think of the best way to handle the checking of "restricted" if we do it this way, but the throttling check is actually pretty easy.

https://github.com/adigitalmonk/durendal-bot/tree/commands-refactor-globals

**Comment by adigitalmonk** [comment]
We could just leave the way we are handling restricted as is (it's a bit more functional because of it), then this would work better.  >_>

**Comment by adigitalmonk** [comment]
This was implemented/merged with #54
